spark-mongodb/src/main/scala/com/stratio/datasource/mongodb/reader/MongodbReader.scala

```diff
@@ -141,6 +141,25 @@ class MongodbReader(
   }
 
   /**
+   * Check if the field is "_id" and if the user wants to filter by this field as an ObjectId
+   *
+   * @param attribute Name of the file
+   * @param value Value for the attribute
+   * @return The value in the correct data type
+   */
+  private def checkObjectID(attribute: String, value: Any) : Any = attribute  match {
+    case "_id" if idAsObjectId => new ObjectId(value.toString)
+    case _ => value
+  }
+
+  private def idAsObjectId: Boolean =
```

Comment:
No need of `if`: `private def idAsObjectId: Boolean = config.getOrElse[String](MongodbConfig.IdAsObjectId, MongodbConfig.DefaultIdAsObjectId).equalsIgnoreCase("true")